FRENCH/src/ch07-00-managing-growing-projects-with-packages-crates-and-modules.md

```diff
@@ -0,0 +1,120 @@
+separate crates that become external dependencies. This chapter covers all
+these techniques. For very large projects of a set of interrelated packages
+that evolve together, Cargo provides workspaces, which we’ll cover in the
+[“Cargo Workspaces”][workspaces]<!-- ignore -- > section in Chapter 14.
+-->
+
+Le programme que nous avons écrit précédemment était dans un module et dans un
+seul fichier. Quand le projet grandit, vous pouvez organiser votre code en le
+découpant en plusieurs modules et ensuite en plusieurs fichiers. Un paquet peut
+contenir plusieurs crates binaires et accessoirement une crate de bibliothèque.
+Lorsque un paquet grandit, vous pouvez en extraire des parties dans des crates
+séparées qui deviennent des dépendances externes. Ce chapitre va aborder toutes
+ces techniques. Pour un projet de très grande envergure qui a des paquets
+interconnectés qui évoluent ensemble, Cargo propose les espaces de travail, que
+nous allons découvrir dans une section du
+[chapitre 14][workspaces]<!-- ignore -->.
+
+<!--
+In addition to grouping functionality, encapsulating implementation details
+lets you reuse code at a higher level: once you’ve implemented an operation,
+other code can call that code via the code’s public interface without knowing
+how the implementation works. The way you write code defines which parts are
+public for other code to use and which parts are private implementation details
+that you reserve the right to change. This is another way to limit the amount
+of detail you have to keep in your head.
+-->
+
+En plus de pouvoir regrouper des fonctionnalités, l'encapsulation des éléments
+implémentés vous permet de réutiliser votre code à un niveau supérieur : lorsque
+vous avez implémenté une opération, un autre code peut appeler ce code via
+l'interface de programmation publique du code sans se soucier de connaître les
+détails de son implémentation. La façon dont vous écrivez le code définit
+quelles sont les parties sont publiques pour qu'un autre code puisse l'utiliser
+et quelles parties sont des détails d'implémentation privées dont vous vous
+réservez le droit de modifier. C'est un autre moyen de limiter la quantité de
+détails que vous devez garder à l'esprit.
+
+<!--
+A related concept is scope: the nested context in which code is written has a
+set of names that are defined as “in scope.” When reading, writing, and
+compiling code, programmers and compilers need to know whether a particular
+name at a particular spot refers to a variable, function, struct, enum, module,
+constant, or other item and what that item means. You can create scopes and
+change which names are in or out of scope. You can’t have two items with the
+same name in the same scope; tools are available to resolve name conflicts.
+-->
+
+Un concept qui lui est associé est la portée : le contexte dans lequel le code
+est écrit a un jeu de noms qui sont définis comme “dans la portée”. Quand ils
```

Comment:
```suggestion
est écrit a un jeu de noms qui sont définis comme “à portée”. Quand ils
```

Reply:
I think `dans la portée` is better because it's a notion of "aera" described in ch04-02.